### PR TITLE
Refactor autocomplete code

### PR DESCRIPTION
**Pull request by cdrini**
Decided to do this as a separate PR first to make it easier to review.

Things to note:
- There could be a styling issue if something includes `master.css` without `jquery.autocomplete.css`, but I couldn't find anywhere where that would matter. I've left more general styling in `jquery.autocomplete.css`, and moved author/language specific styling to `master.css`
- `openlibrary/templates/type/edition/edit.html` uses the old `setup_autocomplete` method, but I can't find anywhere this template is being used

### CODE REVIEW

File: openlibrary/templates/books/edit/edition.html [hunk 1]
```diff
@@ -10,54 +10,31 @@
...
+    </div>
+
+$# Render the ith language input field
 $jsdef render_language(i, language):
-    <div class="input" id="languages-input-$i">
+    <div class="input">
       <input name="languages--$i" class="language language-autocomplete" type="text" id="language-$i" value="$language.name" style="width:300px!important;"/>
       <input name="edition--languages--$i--key" type="hidden" id="language-$i-key" value="$language.key" />
       <a href="javascript:;" class="remove red plain hidden" title="$_('Remove this language')">[x]</a>
-      <br/><a href="javascript:;" class="add small librarian hidden">Add another language?</a>
```
Comment (tfmorris) [inline openlibrary/templates/books/edit/edition.html]: I didn't see any comments about removing this librarian-only hidden functionality. Is it an unrelated change that go mixed in?

Comment (cdrini) [reply openlibrary/templates/books/edit/edition.html]: Good catch! Forgot to bring that up.

I had to change that functionality so that the authors autocomplete and the works autocomplete could share more code. The JS in `openlibrary/plugins/openlibrary/js/autocomplete.js` handles showing/hiding the "Add another _?" link (since it's added for *each* input; not just once) ([relevant code](https://github.com/internetarchive/openlibrary/pull/660/files#diff-3700470797e1cb44772be02ec58c24a7R93)). The language's autocomplete previously had its own JS to handle this correctly ([relevant code](https://github.com/internetarchive/openlibrary/pull/660/files#diff-9d682478ff30e3d66b45bfa90ac289f8L32)), but I figure the benefits of code sharing outweighed the minor change in functionality. This could be refactored further to be flexible enough to accommodate that, but if I recall it would require a decently large change in how the librarian features are shown/hidden, so I decided that could be done another time.

The "Add another language?" functionality is now visible/accessible by default (you no longer have to enable the librarian view).

**Comment by cdrini** [comment]
(Rebased with master)

**Comment by mekarpeles** [comment]
lgtm!
